nncf/experimental/post_training/graph/model_transformer.py

```diff
@@ -0,0 +1,129 @@
+
+        # Transformation commands list
+        self._output_insertion_command = None
+        self._quantizer_insertion_command = None
+
+        self._callbacks_by_commands = {
+            self._output_insertion_command: self._apply_output_insertion_transformations,
+            self._quantizer_insertion_command: self._apply_quantizer_insertion_transformations,
+        }
+
+    def set_model(self, model: ModelType):
+        """
+        Sets model
+
+        :param model: input model
+        """
+        self._model = model
+
+    def prepare_model_for_statistics_collection(self, statistic_points: StatisticPointsContainer) -> ModelType:
+        """
+        Prepares model for statics collection by adding external outputs
+
+        :param statistic_points: StatisticPointsContainer
+        :return: model after transformations
+        """
+        transformation_layout = self._get_transformation_layout_extra_outputs(statistic_points)
+        return self.transform(transformation_layout)
+
+    def transform(self, transformation_layout: TransformationLayout) -> ModelType:
+        """
+        Applies transformation layout on the model
+
+        :param transformation_layout: TransformationLayout
+        :return: model after transformations
+        """
+        commands = list(self._callbacks_by_commands.keys())
+        transformations_list = []
+        for transformation in transformation_layout.transformations:
+            if type(transformation) in commands:
+                transformations_list.append(transformation)
+        self._apply_transformations(transformations_list)
+        return self._model
+
+    def _get_transformation_layout_extra_outputs(self,
+                                                 statistic_points: StatisticPointsContainer) -> TransformationLayout:
+        """
+        Collects transformations layout by statistic_points
+
+        :param statistic_points: StatisticPointsContainer
+        :return: transformation_layout
+        """
+        transformation_layout = self._transformation_layout()
+        transformation_commands = []
+        for _statistic_points in statistic_points.values():
+            for _statistic_point in _statistic_points:
+                transformation_commands.append(
+                    self._output_insertion_command(_statistic_point.target_point))
+
+        for transformation_command in transformation_commands:
+            transformation_layout.register(transformation_command)
+
+        return transformation_layout
+
+    def _apply_transformations(self, transformations: List[TransformationCommand]):
+        """
+        Applies transformations by type-callback on the model
+
+        :param transformations: lisf of the TransformationCommand transformations
+        """
+        transformations_by_types = {c: [] for c in self._callbacks_by_commands}
+        for transformation in transformations:
+            transformation_type = type(transformation)
```

Comment:
still don't understand this. Why you can't just check for the common type here and call corresponding function?

Reply:
To avoid multiple `if's` creation in the future.

Reply:
but anyway if you add new Transformation you should add new mapping to `self._callbacks_by_commands`. This is the same for me.
Your code does the same work but it looks harder than simple if-s statement

Reply:
This logic was changed. Please, feel to comment on the new one.